packages/sdk/src/client.ts

```diff
@@ -688,32 +690,44 @@ export class Client
     async createMediaStream(
         channelId: string | Uint8Array | undefined,
         spaceId: string | Uint8Array | undefined,
+        userId: string | undefined,
         chunkCount: number,
         streamSettings?: PlainMessage<StreamSettings>,
     ): Promise<{ streamId: string; prevMiniblockHash: Uint8Array }> {
         assert(this.userStreamId !== undefined, 'userStreamId must be set')
-        if (!channelId) {
-            assert(spaceId !== undefined, 'spaceId must be set')
+        if (!channelId && !spaceId && !userId) {
+            throw Error('channelId, spaceId or userId must be set')
+        }
+        if (spaceId) {
             assert(isSpaceStreamId(spaceId), 'spaceId must be a valid streamId')
-        } else {
+        }
+        if (channelId) {
             assert(
                 isChannelStreamId(channelId) ||
                     isDMChannelStreamId(channelId) ||
                     isGDMChannelStreamId(channelId),
                 'channelId must be a valid streamId',
             )
+
+            if (!spaceId) {
+                assert(spaceId !== undefined, 'spaceId must be set')
+            }
+        }
+        if (userId) {
+            assert(isUserId(userId), 'userId must be a valid userId')
         }
 
         const streamId = makeUniqueMediaStreamId()
 
-        this.logCall('createMedia', channelId ?? spaceId, streamId)
-
+        const userStreamId = userId ? makeUserStreamId(userId) : undefined
+        this.logCall('createMedia', channelId ?? spaceId ?? userStreamId, streamId)
         const inceptionEvent = await makeEvent(
             this.signerContext,
             make_MediaPayload_Inception({
                 streamId: streamIdAsBytes(streamId),
                 channelId: channelId ? streamIdAsBytes(channelId) : undefined,
                 spaceId: spaceId ? streamIdAsBytes(spaceId) : undefined,
+                userId: userStreamId ? streamIdAsBytes(userStreamId) : undefined,
```

Comment:
When doing some local tests, I found that if I pass `userId` in bytes here, that would be used to create the stream and, of course, would fail. 

Assuming that all the node work is done, and the implementation is correct, the node is expecting a stream_id from this field, so this should be called `userStreamId` instead?

I added a few TODO's in this PR related to this, I'll clean it later.

Reply:
We want to pass userId. I messed up in our demo, we want to use the `requireUserAddr` check instead